pkg/base1/test-external.js

```diff
@@ -44,18 +44,15 @@ QUnit.test("external fsread1", async assert => {
     }));
 
     const req = new XMLHttpRequest();
-    req.open("GET", "/cockpit/channel/" + cockpit.transport.csrf_token + '?' + query);
-    req.onreadystatechange = function() {
-        if (req.readyState == 4) {
-            assert.equal(req.status, 200, "got right status");
-            assert.equal(req.statusText, "OK", "got right reason");
-            assert.equal(req.getResponseHeader("Content-Type"), "application/octet-stream", "default type");
-            assert.equal(req.getResponseHeader("Content-Disposition"), 'attachment; filename="foo"', "default type");
-            assert.equal(req.getResponseHeader("Content-Length"), parseInt(filesize), "expected file size");
-            done();
-        }
-    };
+    req.open("GET", "/cockpit/channel/" + cockpit.transport.csrf_token + '?' + query, false);
     req.send();
+
+    assert.equal(req.status, 200, "got right status");
```

Comment:
Could you please split out that change into a separate commit? At first sight I don't see how that would work? That's an XHR request, which is asynchronous. How can this be done immediately? Does `.send()` block until the reply comes back? That'd be *very* un-JS-y.

Reply:
done